apps/rush-lib/src/logic/operations/OperationSelector.ts

```diff
@@ -3,176 +3,95 @@
 
 import type { RushConfigurationProject } from '../../api/RushConfigurationProject';
 import type { IPhase } from '../../api/CommandLineConfiguration';
-import type { Operation } from './Operation';
+import type { IOperationRunner } from './IOperationRunner';
+
+import { Operation } from './Operation';
+import { OperationStatus } from './OperationStatus';
+import { NullOperationRunner } from './NullOperationRunner';
 
 export interface IOperationSelectorOptions {
   phasesToRun: ReadonlySet<IPhase>;
 }
 
 export interface ICreateOperationsOptions {
+  phaseSelection: ReadonlySet<IPhase>;
   projectSelection: ReadonlySet<RushConfigurationProject>;
-  operationFactory: IOperationFactory;
+  operationFactory: IOperationRunnerFactory;
 }
 
 export interface IOperationOptions {
   project: RushConfigurationProject;
   phase: IPhase;
 }
 
-export interface IOperationFactory {
-  createTask(options: IOperationOptions): Operation;
-}
-
-interface IOperationDependencies {
-  operations: Set<Operation> | undefined;
-  isCacheWriteAllowed: boolean;
-}
-
-interface IOperationNode {
-  key: string;
-  phase: IPhase;
-  project: RushConfigurationProject;
-}
-
-interface ISelectedOperationNode extends IOperationNode {
-  operation: Operation;
+export interface IOperationRunnerFactory {
+  createOperationRunner(options: IOperationOptions): IOperationRunner;
 }
 
 /**
- * This class is responsible for transforming a set of selected phases and selected projects into an operation dependency graph.
+ * This function creates operations for the set of selected projects and phases, using the provided factory
  */
-export class OperationSelector {
-  private readonly _phasesToRun: ReadonlySet<IPhase>;
+export function createOperations(
```

Comment:
The reason for this function signature is for use with `phasedScriptAction.hooks.createOperations.tap('default', createOperations)` and `phasedScriptAction.hooks.createDeltaOperations.tap('default', createOperations)` when those lifecycle hooks are added.